README.tfdoc.hcl

```diff
@@ -0,0 +1,1265 @@
+            type        = string
+            default     = "#{path}"
+            description = <<-END
+              Absolute path, starting with the leading `"/"`. This component is
+              not percent-encoded. The path can contain `#{host}`, `#{path}`,
+              and `#{port}`.
+            END
+          }
+
+          attribute "port" {
+            type        = string
+            default     = "#{port}"
+            description = <<-END
+              The port. Specify a value from `1` to `65535` or `#{port}`.
+            END
+          }
+
+          attribute "protocol" {
+            type        = string
+            default     = "#{protocol}"
+            description = <<-END
+              The protocol. Valid values are `HTTP`, `HTTPS`, or `#{protocol}`. 
+            END
+          }
+
+          attribute "query" {
+            type        = string
+            default     = "#{query"
+            description = <<-END
+              The query parameters, URL-encoded when necessary, but not
+              percent-encoded. Do not include the leading "?".
+            END
+          }
+        }
+      }
+
+      variable "tags" {
+        type        = map(string)
+        default     = {}
+        description = <<-END
+          A map of tags to apply to the `aws_lb_listener` resource.
+        END
+      }
+    }
+
+    section {
+      title = "Extended Resource Configuration"
+
+      variable "rules" {
+        type        = list(rule)
+        default     = []
+        description = <<-END
+          A list of rules to be attached to the created listener.
+        END
+
+        attribute "id" {
+          required    = true
+          type        = string
+          description = <<-END
+            A unique identifier for the rule
+          END
+        }
+
+        attribute "priority" {
+          required    = true
+          type        = number
+          description = <<-END
+            The priority for the rule between `1` and `50000`. Leaving it unset
+            will automatically set the rule with next available priority after
+            currently existing highest rule. A listener can't have multiple
+            rules with the same priority.
+          END
+        }
+
+        attribute "action" {
+          required = true
+          type     = object(action)
+          default = {
+            fixed_response = {
+              content_type = "plain/text"
+              message_body = "Nothing to see here!"
+              status_code  = 418
+            }
+          }
+          description = <<-END
+            Configuration block for default actions.
+          END
+
+          attribute "type" {
+            required    = true
+            type        = string
+            description = <<-END
+              Type of routing action. Valid values are `forward`, `redirect`,
+              `fixed-response`, `authenticate-cognito` and `authenticate-oidc`.
+            END
+          }
```

Comment:
```suggestion
```

This has been abstracted by the sub module in https://github.com/mineiros-io/terraform-aws-lb-listener/blob/main/modules/terraform-aws-lb-listener-rule/main.tf